content/protocols/mqtt.textile

```diff
@@ -0,0 +1,178 @@
+---
+title: MQTT
+meta_description: "Any MQTT-enabled client can communicate with the Ably service through the Ably MQTT protocol adapter. This is especially useful where an Ably SDK is not available for your language of choice."
+languages:
+ - javascript
+ - nodejs
+---
+
+The Ably MQTT protocol adapter is able to translate back and forth between "MQTT":https://mqtt.org/ and Ably's own protocol, allowing for seamless integration of any systems you may have. MQTT (MQ Telemetry Transport) is a "publish/subscribe":https://ably.com/topic/pub-sub, lightweight messaging protocol designed for constrained devices and low-bandwidth networks. One of the major uses of MQTT is with IoT (Internet of Things), where these principles are key to having effective communication between various devices.
+
+h2(#when-to-use). When to use the MQTT adapter
+
+The MQTT adapter is our recommended way of interacting with Ably from devices which do not have a native Ably SDK, such as Arduino platforms, C/C++ applications, and so on. Anyone who has previously been using Pubnub or Pusher SDKs for this purpose may want to consider switching to MQTT. Compared to the Pubnub protocol, using MQTT will result in better performance with lower latency. Compared to the Pusher protocol, MQTT will give you connection state recovery.
+
+MQTT is recommended to interact with Ably when:
+
+* Bandwidth is limited and you want to keep network traffic to a minimum
+* You want to keep network traffic to a minimum.
+
+h2(#limitations). Known limitations 
+
+A protocol adapter inevitably adds some latency. Using the adapter will be a little slower than using Ably client libraries. 
+Typically the impact is in the low milliseconds.
+
+Behind the scenes, the adapter just uses the normal Ably service, so there is no problem with using MQTT and Ably client libraries side by side. You can mix and match as you like; for example, using MQTT on your IoT devices, but using the Ably Realtime API on your servers.
+
+While the adapter can be useful for devices which need to use MQTT, there are many benefits to using the Ably client libraries (e.g. continuity guarantees, fallback host support, history, presence) require the use of the Ably client libraries. As a result, if an Ably client library is available for your platform, we recommend you consider using it instead of the protocol adapter.
```

Comment:
```suggestion
While the adapter can be useful for devices which need to use MQTT, there are many benefits to using an Ably SDK, such as continuity guarantees, fallback host support, history and presence. As a result, if an Ably SDK is available for your platform, it is recommend you consider using it instead of the protocol adapter.
```